luigi/task.py

```diff
@@ -324,7 +337,19 @@ def __hash__(self):
         return self.__hash
 
     def __repr__(self):
-        return self.task_id
+        params = self.get_params()
+        param_values = self.get_param_values(params, [], self.param_kwargs)
+
+        # Build up task id
+        task_id_parts = []
+        param_objs = dict(params)
+        for param_name, param_value in param_values:
+            if param_objs[param_name].significant:
+                task_id_parts.append('%s=%s' % (param_name, param_objs[param_name].serialize(param_value)))
```

Comment:
I believe this line should now be

```
task_id_parts.append('%s=%r' % (param_name, param_value)) 
```

Note the second is %r not %s. Because According to this [SO answer](http://stackoverflow.com/a/2626364/621449), one wants to be close to the `eval(repr(x)) == x` property.